test/e2e/network/util.go

```diff
@@ -141,6 +141,36 @@ func execSourceIPTest(sourcePod v1.Pod, targetAddr string) (string, string) {
 	return sourcePod.Status.PodIP, host
 }
 
+// execHostnameTest executes curl to access "/hostname" endpoint on target address
+// from given Pod to check the hostname of the target destination.
+func execHostnameTest(sourcePod v1.Pod, targetAddr string) string {
```

Comment:
we should pass the hostname and verify that the returned hostname matches the actual hostname

Reply:
That happens from the e2e test, not the util helper. I guess we can do the check in the util function instead :thinking:

Reply:
ah, right, it can return the stdout and we can check from the e2e ... or add it as a parameter 
```
execHostnameTest(sourcePod v1.Pod, targetAddr, targetHostname string)
```

Reply:
yeah, I followed the pattern we used for `execSourceIPTest` where the source IP is returned and checked afterwards.

Reply:
changed my mind, will do the assertion inside execHostnameTest